### ISSUE TITLE
Option for hiding imported items and other noises from intelligence of a Module?

### ISSUE THREAD

**Issue by Shuenhoy** [julia-vscode/julia-vscode#2344]
Julia allows us to use `B.A` if `A` is imported/used into `B`. However, in most cases, if we really want to use `A`, we would directly import/using it into the current scope. And lots of imported items make it hard to find what we actually want.
![image](https://user-images.githubusercontent.com/2196156/128287475-9016c49e-caec-415f-9b33-6626fb08be19.png)

The Julia REPL indeed hides such imported/used items when you use tab key to complete:
![image](https://user-images.githubusercontent.com/2196156/128286494-1f546fd5-c4c0-46b3-88de-e77d49a390c0.png)

The same applies to the module self e.g `B.B`, Julia allows us to write that, but we usually won't do that.

It would help if there is an option to hide these noises (or even hide them directly if no one prefers seeing them).

**Comment by pfitzseb** [comment]
Yeah, I don't think an option is the right way forward here; we should just improve auto-completion suggestions.
That said, it's not entirely clear which behavior is better: I've gotten annoyed at the REPL's completions when a package basically re-exports symbols from a submodule.
